components/TextFields/src/ContainedInputView/private/MDCTextControlAssistiveLabelViewLayout.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "MDCTextControl.h"
+#import "MDCTextControlAssistiveLabelDrawPriority.h"
+
+@interface MDCTextControlAssistiveLabelViewLayout : NSObject
```

Comment:
Lets add a class doc here. This is my understanding of the purpose of this object:
This functional object will compute layout and hold the results of that computation in its readonly properties. 

Other ways that we can do something like this:
https://stackoverflow.com/questions/1692005/returning-multiple-values-from-a-method-in-objective-c

Reply:
Done.